data-pipeline/device-profile-updater/src/main/java/org/ekstep/ep/samza/service/DeviceProfileUpdaterService.java

```diff
@@ -126,10 +129,17 @@ private void addToCache(String deviceId, DeviceProfile deviceProfile, Jedis redi
             if(data.get("firstaccess") != null && !("0").equals(data.get("firstaccess"))) {
                 deviceMap.remove("firstaccess");
             }
+            if(data.get("user_declared_on") == null && data.get("user_declared_state") != null) {
+                deviceMap.put("user_declared_on", String.valueOf(System.currentTimeMillis()));
```

Comment:
This should use the same value as "lastUpdatedDate" similar to the postgres update.

Reply:
Done.